src/main/scala/org/broadinstitute/dsde/workbench/sam/google/DisableUsersMonitor.scala

```diff
@@ -0,0 +1,189 @@
+  import context._
+
+  self ! Init
+
+  override def receive = {
+    case Init => init pipeTo self
+    case Start => for (i <- 1 to workerCount) startOne()
+    case Status.Failure(t) => logger.error("error initializing disable users monitor", t)
+  }
+
+  def init =
+    for {
+      _ <- pubSubDao.createSubscription(pubSubTopicName, pubSubSubscriptionName)
+    } yield Start
+
+  def startOne(): Unit = {
+    logger.info("starting DisableUsersMonitorActor")
+    actorOf(DisableUsersMonitor.props(pollInterval, pollIntervalJitter, pubSubDao, pubSubSubscriptionName, userService))
+  }
+
+  override val supervisorStrategy =
+    OneForOneStrategy() {
+      case e =>
+        logger.error("unexpected error in disable users monitor", e)
+        // start one to replace the error, stop the errored child so that we also drop its mailbox (i.e. restart not good enough)
+        startOne()
+        Stop
+    }
+
+}
+
+object DisableUsersMonitor {
+  case object StartMonitorPass
+
+  sealed abstract class DisableUserResult(ackId: String)
+  final case class ReportMessage(value: Map[WorkbenchEmail, Seq[SyncReportItem]], ackId: String) extends DisableUserResult(ackId = ackId)
+  final case class FailToSynchronize(t: Throwable, ackId: String) extends DisableUserResult(ackId = ackId)
+
+  def props(
+             pollInterval: FiniteDuration,
+             pollIntervalJitter: FiniteDuration,
+             pubSubDao: GooglePubSubDAO,
+             pubSubSubscriptionName: String,
+             userService: UserService
+           ): Props =
+    Props(new DisableUsersMonitorActor(pollInterval, pollIntervalJitter, pubSubDao, pubSubSubscriptionName, userService))
+}
+
+class DisableUsersMonitorActor(
+  val pollInterval: FiniteDuration,
+  pollIntervalJitter: FiniteDuration,
+  pubSubDao: GooglePubSubDAO,
+  pubSubSubscriptionName: String,
+  userService: UserService
+) extends Actor with LazyLogging with FutureSupport {
+  import GoogleGroupSyncMonitor._
+  import context._
+
+  self ! StartMonitorPass
+
+  // fail safe in case this actor is idle too long but not too fast (1 second lower limit)
+  setReceiveTimeout(max((pollInterval + pollIntervalJitter) * 10, 1 second))
+
+  private def max(durations: FiniteDuration*): FiniteDuration = {
+    implicit val finiteDurationIsOrdered = scala.concurrent.duration.FiniteDuration.FiniteDurationIsOrdered
+    durations.max
+  }
+
+  override def receive = {
+    case StartMonitorPass =>
+      // start the process by pulling a message and sending it back to self
+      pubSubDao.pullMessages(pubSubSubscriptionName, 1).map(_.headOption) pipeTo self
+
+    case Some(message: PubSubMessage) =>
+      logger.debug(s"received disable user message: $message")
+      import Tracing._
+      trace("DisableUsersMonitor-PubSubMessage") { span =>
+        userService
+          .disableUser(message.contents.parseJson.convertTo[WorkbenchUserId], samRequestContext = SamRequestContext(Option(span)))
+      }
```

Comment:
I think you are missing something here to message back to the actor to ack the message. Something ending in `pipeTo self`